src/tax/basis.test.js

```diff
@@ -0,0 +1,67 @@
+// @flow
+
+import moment from "moment";
+import type {BasisFrame} from "./basis";
+import {LIFOCostBasisCalculator} from "./basis";
+
+describe("cost basis calculation", () => {
+  const dayFromYear = (x) =>
+    moment().set({
+      year: x,
+      month: 0,
+      day: 0,
+      hour: 0,
+      minute: 0,
+      second: 0,
+      millisecond: 0,
+    });
+  const date0 = dayFromYear(1990);
+  const date1 = dayFromYear(1991);
+  const date2 = dayFromYear(1992);
+
+  it("works in a trivial case", () => {
+    const calc = new LIFOCostBasisCalculator("FOO");
+    calc.acquire(date0, 1, 100);
+    const results = calc.dispose(1);
+    expect(results).toEqual([{amount: 1, cost: 100, date: date0}]);
+  });
+
+  it("uses LIFO semantics", () => {
+    const calc = new LIFOCostBasisCalculator("FOO");
+    calc.acquire(date0, 1, 100);
+    calc.acquire(date1, 1, 200);
+    const results = calc.dispose(1.5);
+    expect(results).toEqual([
+      {amount: 1, cost: 200, date: date1},
+      {amount: 0.5, cost: 100, date: date0},
+    ]);
+  });
+
+  it("throws error when disposing without acquiring", () => {
+    const calc = new LIFOCostBasisCalculator("FOO");
+    expect(() => calc.dispose(3)).toThrow("none remained");
```

Comment:
Huh. The fact that this passes indicates that Jest’s docs are wrong: they say that a string argument to `toThrow` tests “the exact error message”, while you should use a regex to test that the error contains the text “somewhere”. But indeed the source both [at your version of Jest][1] and [at current master][2] clearly indicates that this is not the behavior. Maybe I’ll PR the docs.

[1]: https://github.com/facebook/jest/blob/c27b759f256adde5586b0c69e510a3aa81d0f869/packages/jest-matchers/src/toThrowMatchers.js#L51-L53
[2]: https://github.com/facebook/jest/blob/5a21be3e07c223ba9fbb434f26bce96da0dddd31/packages/expect/src/to_throw_matchers.js#L53-L55

Reply:
Ack